src/R/Components/Impl/PackageManager/Implementation/ViewModel/RPackageViewModel.cs

```diff
@@ -0,0 +1,149 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using Microsoft.Common.Core;
+using Microsoft.Common.Wpf;
+using Microsoft.R.Components.PackageManager.Model;
+using Microsoft.R.Components.PackageManager.ViewModel;
+
+namespace Microsoft.R.Components.PackageManager.Implementation.ViewModel {
+    internal class RPackageViewModel : BindableBase, IRPackageViewModel {
+        private bool _hasDetails;
+        private bool _isSelected;
+        private bool _isInstalled;
+        private string _title;
+        private string _description;
+        private string _authors;
+        private string _installedVersion;
+        private string _latestVersion;
+        private ICollection<string> _urls;
+        private string _libraryPath;
+        private string _built;
+
+        public static RPackageViewModel CreateAvailable(RPackage package) {
+            return new RPackageViewModel(package.Package) {
+                LatestVersion = package.Version,
+                Depends = package.Depends,
+                Imports = package.Imports,
+                Suggests = package.Suggests,
+                License = package.License,
+                NeedsCompilation = package.NeedsCompilation != null && !package.NeedsCompilation.EqualsIgnoreCase("no"),
+                Repository = package.Repository,
+            };
+        }
+
+        public static RPackageViewModel CreateInstalled(RPackage package) {
+            return new RPackageViewModel(package.Package) {
+                Title = package.Title,
+                Authors = package.Author,
+                License = package.License,
+                Urls = package.URL?.Split(new[] { "," }, StringSplitOptions.RemoveEmptyEntries)
```

Comment:
Because I might not want a `null` to `null` mapping on the next step. For example, my extension method might be something like:
```
string Stringify(this Foo foo) {
  if (foo == null) return "<null>"
  ...
}
```
And then if I do `bar?.GetFoo().Stringify()`, I would expect to get `<null>` if `bar` was null; but I  get actual null instead, because `Stringify()` isn't called at all. I'd prefer to have full control on every step by using `.` or `?.` as necessary.

(I recall now that there was actually a discussion about this back when they were still designing the feature.)